includes/class-wcpdf-main.php

```diff
@@ -257,12 +257,32 @@ public function get_documents_for_email( $email_id, $order ) {
 	 */
 	public function generate_pdf_ajax() {
 		$guest_access = isset( WPO_WCPDF()->settings->debug_settings['guest_access'] );
-		if ( !$guest_access && current_filter() == 'wp_ajax_nopriv_generate_wpo_wcpdf') {
+		if ( ! $guest_access && current_filter() == 'wp_ajax_nopriv_generate_wpo_wcpdf' ) {
+			wp_die( esc_attr__( 'You do not have sufficient permissions to access this page.', 'woocommerce-pdf-invoices-packing-slips' ) );
+		}
+
+		// check if the access type is 'order_key'
```

Comment:
is this necessary? You can't be 100% certain that the `order_key` will start with `wc_order_`, because it can be filtered ([source](https://github.com/woocommerce/woocommerce/blob/e62a338826d0b15b23a80fced793ee86e888ffd5/plugins/woocommerce/includes/wc-order-functions.php#L158-L164)).
I think you can simply check if it validates as a nonce first, and if not, and gues access is enabled, you can check if it validates as the order key.

Reply:
Ok, makes sense.